File: Lab9/app/src/main/java/psy/fit/bstu/lab9/adapters/ViewAdapter.java
package psy.fit.bstu.lab9.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import psy.fit.bstu.lab9.Contact;
import psy.fit.bstu.lab9.R;

public class ViewAdapter extends RecyclerView.Adapter<ViewAdapter.ViewHolder> {
    private LayoutInflater inflater;
    private List<Contact> contacts;

    @NotNull
    @Override
    public ViewAdapter.ViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ViewHolder(view);
    }
    @Override
    public void onAttachedToRecyclerView(@NotNull RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
    }
    @Override
    public void onBindViewHolder(@NotNull ViewAdapter.ViewHolder holder, int position) {
        if(contacts != null) {
            Contact phone = contacts.get(position);
            holder.nameView.setText(phone.getName());
            holder.phoneView.setText(phone.getPhone());
        }
    }

    @Override
    public int getItemCount() {
        return contacts.size();
    }

    public void setContacts(List<Contact> contacts) {
        this.contacts = contacts;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        final TextView nameView, phoneView;

        ViewHolder(View view) {
            super(view);
            nameView = (TextView) view.findViewById(R.id.idName);
            phoneView = (TextView) view.findViewById(R.id.idPhone);
        }
    }
}

File: lab13/app/src/main/java/psy/fit/bstu/lab13/ViewAdapter.kt
package psy.fit.bstu.lab13

import IBaseModel
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class ViewAdapter(_posts: List<IBaseModel>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var posts: List<IBaseModel> = _posts

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(
            R.layout.rows,
            parent,
            false
        )
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val view = holder.itemView.findViewById(R.id.placeName) as TextView
        view.text = posts[position].toString()
    }

    override fun getItemCount(): Int = posts.size
}